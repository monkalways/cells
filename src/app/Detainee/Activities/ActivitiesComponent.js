import React from 'react';
import PropTypes from 'prop-types';
import {
  Avatar, Grid, Typography, withStyles,
} from '@material-ui/core';

import MedicalVisitIcon from '../../../images/MedicalVisit.png';
import PhoneAcceptIcon from '../../../images/PhoneAccept.png';
import PhoneDeclineIcon from '../../../images/PhoneDecline.png';
import FingerprintingIcon from '../../../images/Fingerprinting.png';
import BreathalyzerIcon from '../../../images/Breathalyzer.png';
import InterviewIcon from '../../../images/Interview.png';
import BailHearingIcon from '../../../images/BailHearingGeneral.png';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  detainee: PropTypes.shape({
    mealAcceptedCount: PropTypes.number.isRequired,
    mealDeclinedCount: PropTypes.number.isRequired,
    medicationAcceptedCount: PropTypes.number.isRequired,
    medicationRefusedCount: PropTypes.number.isRequired,
    verbalCellCheckCount: PropTypes.number.isRequired,
    visualCellCheckCount: PropTypes.number.isRequired,
  }).isRequired,
};

const variant = 'subtitle2';

const ActivitiesComponent = ({ classes, detainee }) => (
  <Grid container className={classes.grid}>
    <Grid item xs={12}>
      <Typography variant="h6" className={classes.text}>
        Activities
      </Typography>
    </Grid>
    <Grid container item xs={12}>
      <Grid item xs={1} className={`${classes.gridItem} ${classes.topLeft}`}>
        <Avatar className={classes.avatar} src={MedicalVisitIcon} />
      </Grid>
      <Grid item xs={1} className={`${classes.gridItem} ${classes.topCenter}`}>
        <Avatar className={classes.avatar} src={PhoneAcceptIcon} />
      </Grid>
      <Grid item xs={1} className={`${classes.gridItem} ${classes.topCenter}`}>
        <Avatar className={classes.avatar} src={PhoneDeclineIcon} />
      </Grid>
      <Grid item xs={1} className={`${classes.gridItem} ${classes.topCenter}`}>
        <Avatar className={classes.avatar} src={FingerprintingIcon} />
      </Grid>
      <Grid item xs={1} className={`${classes.gridItem} ${classes.topCenter}`}>
        <Avatar className={classes.avatar} src={BreathalyzerIcon} />
      </Grid>
      <Grid item xs={1} className={`${classes.gridItem} ${classes.topCenter}`}>
        <Avatar className={classes.avatar} src={InterviewIcon} />
      </Grid>
      <Grid item xs={1} className={`${classes.gridItem} ${classes.topRight}`}>
        <Avatar className={classes.avatar} src={BailHearingIcon} />
      </Grid>
    </Grid>

    <Grid container item xs={12}>
      <Grid item xs={1} className={`${classes.gridItem} ${classes.bottomLeft}`}>
        <Typography variant={variant} align="center">
          {detainee.medicalCount}
        </Typography>
      </Grid>
      <Grid
        item
        xs={1}
        className={`${classes.gridItem} ${classes.bottomCenter}`}
      >
        <Typography variant={variant} align="center">
          {detainee.telephoneAcceptedCount}
        </Typography>
      </Grid>
      <Grid
        item
        xs={1}
        className={`${classes.gridItem} ${classes.bottomCenter}`}
      >
        <Typography variant={variant} align="center">
          {detainee.telephoneDeclinedCount}
        </Typography>
      </Grid>
      <Grid
        item
        xs={1}
        className={`${classes.gridItem} ${classes.bottomCenter}`}
      >
        <Typography variant={variant} align="center">
          {detainee.fingerPrintCount}
        </Typography>
      </Grid>
      <Grid
        item
        xs={1}
        className={`${classes.gridItem} ${classes.bottomCenter}`}
      >
        <Typography variant={variant} align="center">
          {detainee.breathTestCount}
        </Typography>
      </Grid>
      <Grid
        item
        xs={1}
        className={`${classes.gridItem} ${classes.bottomCenter}`}
      >
        <Typography variant={variant} align="center">
          {detainee.interviewCount}
        </Typography>
      </Grid>
      <Grid
        item
        xs={1}
        className={`${classes.gridItem} ${classes.bottomRight}`}
      >
        <Typography variant={variant} align="center">
          {detainee.bailHearingCount}
        </Typography>
      </Grid>
    </Grid>
  </Grid>
);
ActivitiesComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  avatar: {
    height: theme.spacing.unit * 6,
    width: theme.spacing.unit * 6,
  },
  bottomLeft: { borderWidth: '1px 1px 0 0' },
  bottomCenter: { borderWidth: '1px 1px 0 1px' },
  bottomRight: { borderWidth: '1px 0 0 1px' },
  grid: {
    padding: theme.spacing.unit * 0.5,
  },
  gridItem: {
    borderStyle: 'solid',
    borderColor: 'black',
    padding: theme.spacing.unit * 0.5,
  },
  text: {
    textDecoration: 'underline',
  },
  topLeft: { borderWidth: '0 1px 1px 0' },
  topCenter: { borderWidth: '0 1px 1px 1px' },
  topRight: { borderWidth: '0 0 1px 1px' },
}))(ActivitiesComponent);
