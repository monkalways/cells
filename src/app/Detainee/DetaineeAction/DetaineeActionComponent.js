import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import {
  Avatar,
  Card,
  CardHeader,
  CardContent,
  Grid,
  Typography,
  withStyles,
} from '@material-ui/core';
import IconButton from '@material-ui/core/IconButton';

import MedicalVisitIcon from '../../../images/MedicalVisit.png';
import PhoneAcceptIcon from '../../../images/PhoneAccept.png';
import PhoneDeclineIcon from '../../../images/PhoneDecline.png';
import FingerprintingIcon from '../../../images/Fingerprinting.png';
import BreathalyzerIcon from '../../../images/Breathalyzer.png';
import InterviewIcon from '../../../images/Interview.png';
import BailHearing1Icon from '../../../images/BailHearing1.png';
import BailHearing2Icon from '../../../images/BailHearing2.png';
import RemandHoldingIcon from '../../../images/RemandHolding.png';
import ReleaseHoldingIcon from '../../../images/ReleaseHolding.png';
import InCellIcon from '../../../images/InCell.png';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  isMedicalRoomOptionAvailable: PropTypes.bool.isRequired,
  isPhoneRoomOptionAvailable: PropTypes.bool.isRequired,
};

const DetaineeActionComponent = ({
  classes,
  isMedicalRoomOptionAvailable,
  isPhoneRoomOptionAvailable,
}) => (
  <Card className={classes.card}>
    <CardHeader
      component={() => (
        <Typography variant="h6" className={classes.cardHeader}>
          Detainee Action
        </Typography>
      )}
    />
    <CardContent className={classes.cardContent}>
      <Grid container spacing={16}>
        <Grid container item xs={12}>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton
              className={classes.button}
              disabled={!isMedicalRoomOptionAvailable}
            >
              <Avatar
                className={classNames(classes.avatar, {
                  [classes.disabled]: !isMedicalRoomOptionAvailable,
                })}
                src={MedicalVisitIcon}
              />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton
              className={classes.button}
              disabled={!isPhoneRoomOptionAvailable}
            >
              <Avatar
                className={classNames(classes.avatar, {
                  [classes.disabled]: !isPhoneRoomOptionAvailable,
                })}
                src={PhoneAcceptIcon}
              />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={PhoneDeclineIcon} />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={FingerprintingIcon} />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={BreathalyzerIcon} />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={InterviewIcon} />
            </IconButton>
          </Grid>
        </Grid>

        <Grid container item xs={12}>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={BailHearing1Icon} />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={BailHearing2Icon} />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={RemandHoldingIcon} />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={ReleaseHoldingIcon} />
            </IconButton>
          </Grid>
          <Grid item xs={2} className={classes.imgContainer}>
            <IconButton className={classes.button}>
              <Avatar className={classes.avatar} src={InCellIcon} />
            </IconButton>
          </Grid>
        </Grid>
      </Grid>
    </CardContent>
  </Card>
);
DetaineeActionComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  avatar: {
    height: theme.spacing.unit * 6,
    width: theme.spacing.unit * 6,
  },
  button: {
    padding: theme.spacing.unit * 0.5,
  },
  cardContent: {
    backgroundColor: '#d4e2fc',
    paddingBottom: [16, '!important'],
  },
  cardHeader: {
    backgroundColor: '#3d89f7',
    paddingLeft: theme.spacing.unit,
    paddingTop: theme.spacing.unit * 0.5,
    paddingBottom: theme.spacing.unit * 0.5,
  },
  disabled: {
    opacity: '0.2',
  },
  imgContainer: {
    display: 'flex',
    justifyContent: 'center',
  },
}))(DetaineeActionComponent);
