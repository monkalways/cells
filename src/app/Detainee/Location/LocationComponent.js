import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Typography, withStyles } from '@material-ui/core';

import InCellIcon from '../../../images/InCell.png';
import InTransitIcon from '../../../images/InTransit.png';
import BailHearingGeneralIcon from '../../../images/BailHearingGeneral.png';
import FingerprintingIcon from '../../../images/Fingerprinting.png';
import MedicalVisitIcon from '../../../images/MedicalVisit.png';
import PhoneAcceptIcon from '../../../images/PhoneAccept.png';

const LOCATION_PHONE_ACCEPT = 'phone - in progress';
const LOCATION_MEDICAL_VISIT = 'medical - in progress';
const LOCATION_FINGER_PRINTING = 'fingerprinting - in progress';
const LOCATION_BAIL_HEARING = 'bail hearing - in progress';
const LOCATION_IN_TRANSIT = 'in transit';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  location: PropTypes.string.isRequired,
};

const LocationComponent = ({ classes, location }) => {
  const getLocationIconSrc = () => {
    let locationIconSrc = InCellIcon;

    if (location) {
      if (location.toLowerCase().includes(LOCATION_IN_TRANSIT)) {
        locationIconSrc = InTransitIcon;
      }

      switch (location.toLowerCase()) {
        case LOCATION_PHONE_ACCEPT:
          locationIconSrc = PhoneAcceptIcon;
          break;
        case LOCATION_MEDICAL_VISIT:
          locationIconSrc = MedicalVisitIcon;
          break;
        case LOCATION_FINGER_PRINTING:
          locationIconSrc = FingerprintingIcon;
          break;
        case LOCATION_BAIL_HEARING:
          locationIconSrc = BailHearingGeneralIcon;
          break;
        default:
          break;
      }
    }

    return locationIconSrc;
  };

  return (
    <Grid
      container
      direction="column"
      justify="space-between"
      className={classes.column}
      spacing={8}
    >
      <Grid item>
        <Typography>Location</Typography>
      </Grid>
      <Grid item>
        <img
          className={classes.img}
          alt="Detainee Location"
          src={getLocationIconSrc()}
        />
      </Grid>
    </Grid>
  );
};

LocationComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  img: {
    width: '150px',
    height: '150px',
  },
  column: {
    padding: theme.spacing.unit * 0.5,
  },
}))(LocationComponent);
