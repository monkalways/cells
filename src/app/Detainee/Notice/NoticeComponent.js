import React from 'react';
import PropTypes from 'prop-types';
import {
  Avatar, Grid, Typography, withStyles,
} from '@material-ui/core';

import CautionPresentIcon from '../../../images/CautionPresent.png';
import ContagiousIcon from '../../../images/Contagious.png';
import KeepAloneIcon from '../../../images/KeepAlone.png';
import MedicineAcceptIcon from '../../../images/MedicineAccept.png';
import OtherWarningIcon from '../../../images/OtherWarning.png';
import SuicideWarningIcon from '../../../images/SuicideWarning.png';

const CAUTION_PRESENT = 'CautionPresent';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  detainee: PropTypes.shape({
    withCaution: PropTypes.bool.isRequired,
    mustBeKeptAlone: PropTypes.bool.isRequired,
    isSuicidal: PropTypes.bool.isRequired,
    isContagious: PropTypes.bool.isRequired,
    hasWarning: PropTypes.bool.isRequired,
    isUnderMedication: PropTypes.bool.isRequired,
    cautionsArray: PropTypes.arrayOf(PropTypes.string).isRequired,
  }).isRequired,
};

const NoticeComponent = ({ classes, detainee }) => {
  const renderWarnings = () => {
    const icons = [];
    if (detainee.withCaution) icons.push(CautionPresentIcon);
    if (detainee.mustBeKeptAlone) icons.push(KeepAloneIcon);
    if (detainee.isSuicidal) icons.push(SuicideWarningIcon);
    if (detainee.isContagious) icons.push(ContagiousIcon);
    if (detainee.hasWarning) icons.push(OtherWarningIcon);
    if (detainee.isUnderMedication) icons.push(MedicineAcceptIcon);

    return icons.map((icon) => {
      if (icon.includes(CAUTION_PRESENT)) {
        return (
          <Grid item key={icon} xs={1} className={classes.gridItem}>
            <Avatar className={classes.avatar} src={icon} />
            <Typography align="center" className={classes.cautionText}>
              {detainee.cautionsArray.join('')}
            </Typography>
          </Grid>
        );
      }
      return (
        <Grid item key={icon} xs={1} className={classes.gridItem}>
          <Avatar className={classes.avatar} src={icon} />
        </Grid>
      );
    });
  };

  return (
    <Grid container className={classes.grid}>
      <Grid item xs={12}>
        <Typography variant="h6" className={classes.text}>
          Notice
        </Typography>
      </Grid>
      <Grid container item xs={12}>
        {renderWarnings()}
      </Grid>
    </Grid>
  );
};

NoticeComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  avatar: {
    height: theme.spacing.unit * 6,
    width: theme.spacing.unit * 6,
    zIndex: 99,
  },
  cautionText: {
    color: '#FFFFFF',
    backgroundColor: '#F80205',
    width: theme.spacing.unit * 6,
    fontWeight: theme.typography.caption.fontWeight,
    fontSize: theme.typography.caption.fontSize,
  },
  grid: {
    padding: theme.spacing.unit * 0.5,
    minHeight: theme.spacing.unit * 12,
  },
  gridItem: {
    padding: theme.spacing.unit * 0.5,
  },
  text: {
    textDecoration: 'underline',
  },
}))(NoticeComponent);
