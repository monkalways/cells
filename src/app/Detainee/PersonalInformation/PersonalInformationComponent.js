import React from 'react';
import PropTypes from 'prop-types';
import {
  Grid, Paper, Typography, withStyles,
} from '@material-ui/core';
import detaineeImage from './detainee.PNG';
import PersonalStats from './PersonalStats';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  detainee: PropTypes.shape({}).isRequired,
};

const PersonalInformationComponent = ({ classes, detainee }) => {
  console.log(detainee);

  const getDetaineeImage = () => {
    const { intakePhotoResourceUri } = detainee;
    if (intakePhotoResourceUri) {
      return intakePhotoResourceUri;
    }
    return detaineeImage;
  };

  return (
    <React.Fragment>
      <Grid
        container
        direction="column"
        justify="space-between"
        className={classes.column}
        spacing={8}
      >
        <Grid item>
          <Typography>Personal Information</Typography>
        </Grid>
        <Grid item>
          <Grid container alignItems="flex-start" justify="flex-start">
            <Grid item sm={4}>
              <img
                className={classes.img}
                alt="Detainee"
                src={getDetaineeImage()}
              />
            </Grid>
            <Grid item sm={8}>
              <PersonalStats detainee={detainee} />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </React.Fragment>
  );
};

PersonalInformationComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  img: {
    width: '150px',
    height: '150px',
  },
  row: {
    padding: theme.spacing.unit * 0.5,
  },
  column: {
    padding: theme.spacing.unit * 0.5,
  },
}))(PersonalInformationComponent);
