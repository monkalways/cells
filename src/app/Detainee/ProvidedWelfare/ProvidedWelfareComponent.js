import React from 'react';
import PropTypes from 'prop-types';
import {
  Avatar,
  Divider,
  Grid,
  Typography,
  withStyles,
} from '@material-ui/core';

import VisualCheckIcon from '../../../images/WellnessVisual.png';
import VerbalCheckIcon from '../../../images/WellnessVerbal.png';
import MealAcceptIcon from '../../../images/MealAccept.png';
import MealDeclineIcon from '../../../images/MealDecline.png';
import MedicineAcceptIcon from '../../../images/MedicineAccept.png';
import MedicineDeclineIcon from '../../../images/MedicineDecline.png';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  detainee: PropTypes.shape({
    mealAcceptedCount: PropTypes.number.isRequired,
    mealDeclinedCount: PropTypes.number.isRequired,
    medicationAcceptedCount: PropTypes.number.isRequired,
    medicationRefusedCount: PropTypes.number.isRequired,
    verbalCellCheckCount: PropTypes.number.isRequired,
    visualCellCheckCount: PropTypes.number.isRequired,
  }).isRequired,
};

const gridSpacing = 1;
const variant = 'subtitle2';

const ProvidedWelfareComponent = ({ classes, detainee }) => (
  <Grid container spacing={8} className={classes.grid}>
    <Grid item xs={12}>
      <Typography variant="subtitle2">Provided Welfare</Typography>
    </Grid>
    <Grid item xs={12}>
      <Grid container spacing={16}>
        <Grid item xs={gridSpacing}>
          <Avatar className={classes.avatar} src={VisualCheckIcon} />
        </Grid>
        <Grid item xs={gridSpacing}>
          <Avatar className={classes.avatar} src={VerbalCheckIcon} />
        </Grid>
        <Grid item xs={gridSpacing}>
          <Avatar className={classes.avatar} src={MealAcceptIcon} />
        </Grid>
        <Grid item xs={gridSpacing}>
          <Avatar className={classes.avatar} src={MealDeclineIcon} />
        </Grid>
        <Grid item xs={gridSpacing}>
          <Avatar className={classes.avatar} src={MedicineAcceptIcon} />
        </Grid>
        <Grid item xs={gridSpacing}>
          <Avatar className={classes.avatar} src={MedicineDeclineIcon} />
        </Grid>
      </Grid>
    </Grid>
    <Grid item xs={12}>
      <Divider />
    </Grid>
    <Grid item xs={12}>
      <Grid container spacing={16}>
        <Grid item xs={gridSpacing}>
          <Typography variant={variant} align="center">
            {detainee.visualCellCheckCount}
          </Typography>
        </Grid>
        <Grid item xs={gridSpacing}>
          <Typography variant={variant} align="center">
            {detainee.verbalCellCheckCount}
          </Typography>
        </Grid>
        <Grid item xs={gridSpacing}>
          <Typography variant={variant} align="center">
            {detainee.mealAcceptedCount}
          </Typography>
        </Grid>
        <Grid item xs={gridSpacing}>
          <Typography variant={variant} align="center">
            {detainee.mealDeclinedCount}
          </Typography>
        </Grid>
        <Grid item xs={gridSpacing}>
          <Typography variant={variant} align="center">
            {detainee.medicationAcceptedCount}
          </Typography>
        </Grid>
        <Grid item xs={gridSpacing}>
          <Typography variant={variant} align="center">
            {detainee.medicationRefusedCount}
          </Typography>
        </Grid>
      </Grid>
    </Grid>
  </Grid>
);
ProvidedWelfareComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  avatar: {
    height: theme.spacing.unit * 6,
    width: theme.spacing.unit * 6,
  },
  grid: {
    padding: theme.spacing.unit * 0.5,
  },
}))(ProvidedWelfareComponent);
