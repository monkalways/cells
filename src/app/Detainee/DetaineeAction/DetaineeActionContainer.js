import { connect } from 'react-redux';
import { selectors } from '../duck';
import DetaineeActionComponent from './DetaineeActionComponent';

export const mapStateToProps = (
  state,
  isMedicalRoomOptionAvailable = selectors.isMedicalRoomOptionAvailableState(state),
  isPhoneDeclineOptionAvailable = selectors.isPhoneDeclineOptionAvailableState(state),
  isPhoneRoomOptionAvailable = selectors.isPhoneRoomOptionAvailableState(state),
) => ({
  isMedicalRoomOptionAvailable,
  isPhoneDeclineOptionAvailable,
  isPhoneRoomOptionAvailable,
});

export const mapDispatchToProps = () => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(DetaineeActionComponent);
