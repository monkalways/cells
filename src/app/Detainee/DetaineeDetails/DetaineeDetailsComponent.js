import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from '@material-ui/core';
import Loading from '../../common/Loading';
import DetaineeGrid from '../DetaineeGrid';
import PersonalInformation from '../PersonalInformation';
import Location from '../Location';
import Notice from '../Notice';
import ProvidedWelfare from '../ProvidedWelfare';
import Activities from '../Activities';
import DetaineeAction from '../DetaineeAction';

const propTypes = {
  detainee: PropTypes.shape({}),
  isDetaineeProfileLoaded: PropTypes.bool.isRequired,
};

const defaultProps = {
  detainee: null,
};

const DetaineeDetailsComponent = ({ detainee, isDetaineeProfileLoaded }) => (
  <React.Fragment>
    <DetaineeGrid>
      {isDetaineeProfileLoaded ? (
        <Grid container>
          <Grid item sm={10}>
            <PersonalInformation detainee={detainee} />
          </Grid>
          <Grid item sm={2}>
            <Location location={detainee.location} />
          </Grid>
          <Grid item sm={12}>
            <Notice detainee={detainee} />
          </Grid>
          <Grid item sm={12}>
            <ProvidedWelfare detainee={detainee} />
          </Grid>
          <Grid item sm={12}>
            <Activities detainee={detainee} />
          </Grid>
          <Grid item sm={12}>
            <DetaineeAction detainee={detainee} />
          </Grid>
        </Grid>
      ) : (
        <Loading />
      )}
    </DetaineeGrid>
  </React.Fragment>
);

DetaineeDetailsComponent.propTypes = propTypes;
DetaineeDetailsComponent.defaultProps = defaultProps;

export default DetaineeDetailsComponent;
