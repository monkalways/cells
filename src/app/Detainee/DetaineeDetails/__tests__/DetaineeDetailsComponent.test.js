import React from 'react';
import { shallow } from 'enzyme';

import { DetaineeDetailsComponent } from '../DetaineeDetailsComponent';

import PersonalInformation from '../../PersonalInformation';
import Location from '../../Location';
import Notice from '../../Notice';
import ProvidedWelfare from '../../ProvidedWelfare';
import Activities from '../../Activities';
import DetaineeAction from '../../DetaineeAction';
import Loading from '../../../common/Loading';

describe('DetaineeDetailsComponent', () => {
  let detainee;
  let isDetaineeProfileLoaded;

  beforeEach(() => {
    detainee = {
      bailHearingCount: 0,
      breathTestCount: 1,
      cautionsArray: [''],
      detentionUnitName: 'Detainee Management Unit',
      dob: '2001-01-01',
      firstName: 'Steveo',
      fingerPrintCount: 3,
      gender: 'M',
      hasWarning: false,
      id: '123',
      intakePhotoResourceUri: null,
      interviewCount: 4,
      isContagious: false,
      isSuicidal: false,
      isUnderMedication: false,
      lastName: 'Nordic',
      location: '',
      medicalCount: 5,
      mustBeKeptAlone: false,
      propertyBagNumber: null,
      telephoneAcceptedCount: 6,
      telephoneDeclinedCount: 7,
      withCaution: false,
    };
    isDetaineeProfileLoaded = false;
  });

  const setup = () => shallow(<DetaineeDetailsComponent
    detainee={detainee}
    isDetaineeProfileLoaded={isDetaineeProfileLoaded}
  />);

  it('should render if detainee profile is loaded', () => {
    isDetaineeProfileLoaded = true;

    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(PersonalInformation)).toExist();
    expect(wrapper.find(Location)).toExist();
    expect(wrapper.find(Notice)).toExist();
    expect(wrapper.find(ProvidedWelfare)).toExist();
    expect(wrapper.find(Activities)).toExist();
    expect(wrapper.find(DetaineeAction)).toExist();
  });

  it('should render Loading component if detainee profile is not loaded', () => {
    isDetaineeProfileLoaded = false;

    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Loading)).toExist();
  });
});
