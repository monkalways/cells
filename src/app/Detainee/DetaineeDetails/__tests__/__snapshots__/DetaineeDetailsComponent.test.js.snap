// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetaineeDetailsComponent should render Loading component if detainee profile is not loaded 1`] = `
<Fragment>
  <WithStyles(DetaineeGridComponent)>
    <WithStyles(LoadingComponent) />
  </WithStyles(DetaineeGridComponent)>
</Fragment>
`;

exports[`DetaineeDetailsComponent should render if detainee profile is loaded 1`] = `
<Fragment>
  <WithStyles(DetaineeGridComponent)>
    <WithStyles(Grid)
      container={true}
    >
      <WithStyles(Grid)
        item={true}
        sm={10}
      >
        <WithStyles(PersonalInformationComponent)
          detainee={
            Object {
              "bailHearingCount": 0,
              "breathTestCount": 1,
              "cautionsArray": Array [
                "",
              ],
              "detentionUnitName": "Detainee Management Unit",
              "dob": "2001-01-01",
              "fingerPrintCount": 3,
              "firstName": "Steveo",
              "gender": "M",
              "hasWarning": false,
              "id": "123",
              "intakePhotoResourceUri": null,
              "interviewCount": 4,
              "isContagious": false,
              "isSuicidal": false,
              "isUnderMedication": false,
              "lastName": "Nordic",
              "location": "",
              "medicalCount": 5,
              "mustBeKeptAlone": false,
              "propertyBagNumber": null,
              "telephoneAcceptedCount": 6,
              "telephoneDeclinedCount": 7,
              "withCaution": false,
            }
          }
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        sm={2}
      >
        <WithStyles(LocationComponent)
          detainee={
            Object {
              "bailHearingCount": 0,
              "breathTestCount": 1,
              "cautionsArray": Array [
                "",
              ],
              "detentionUnitName": "Detainee Management Unit",
              "dob": "2001-01-01",
              "fingerPrintCount": 3,
              "firstName": "Steveo",
              "gender": "M",
              "hasWarning": false,
              "id": "123",
              "intakePhotoResourceUri": null,
              "interviewCount": 4,
              "isContagious": false,
              "isSuicidal": false,
              "isUnderMedication": false,
              "lastName": "Nordic",
              "location": "",
              "medicalCount": 5,
              "mustBeKeptAlone": false,
              "propertyBagNumber": null,
              "telephoneAcceptedCount": 6,
              "telephoneDeclinedCount": 7,
              "withCaution": false,
            }
          }
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        sm={12}
      >
        <WithStyles(NoticeComponent)
          detainee={
            Object {
              "bailHearingCount": 0,
              "breathTestCount": 1,
              "cautionsArray": Array [
                "",
              ],
              "detentionUnitName": "Detainee Management Unit",
              "dob": "2001-01-01",
              "fingerPrintCount": 3,
              "firstName": "Steveo",
              "gender": "M",
              "hasWarning": false,
              "id": "123",
              "intakePhotoResourceUri": null,
              "interviewCount": 4,
              "isContagious": false,
              "isSuicidal": false,
              "isUnderMedication": false,
              "lastName": "Nordic",
              "location": "",
              "medicalCount": 5,
              "mustBeKeptAlone": false,
              "propertyBagNumber": null,
              "telephoneAcceptedCount": 6,
              "telephoneDeclinedCount": 7,
              "withCaution": false,
            }
          }
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        sm={12}
      >
        <WithStyles(ProvidedWelfareComponent)
          detainee={
            Object {
              "bailHearingCount": 0,
              "breathTestCount": 1,
              "cautionsArray": Array [
                "",
              ],
              "detentionUnitName": "Detainee Management Unit",
              "dob": "2001-01-01",
              "fingerPrintCount": 3,
              "firstName": "Steveo",
              "gender": "M",
              "hasWarning": false,
              "id": "123",
              "intakePhotoResourceUri": null,
              "interviewCount": 4,
              "isContagious": false,
              "isSuicidal": false,
              "isUnderMedication": false,
              "lastName": "Nordic",
              "location": "",
              "medicalCount": 5,
              "mustBeKeptAlone": false,
              "propertyBagNumber": null,
              "telephoneAcceptedCount": 6,
              "telephoneDeclinedCount": 7,
              "withCaution": false,
            }
          }
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        sm={12}
      >
        <WithStyles(ActivitiesComponent)
          detainee={
            Object {
              "bailHearingCount": 0,
              "breathTestCount": 1,
              "cautionsArray": Array [
                "",
              ],
              "detentionUnitName": "Detainee Management Unit",
              "dob": "2001-01-01",
              "fingerPrintCount": 3,
              "firstName": "Steveo",
              "gender": "M",
              "hasWarning": false,
              "id": "123",
              "intakePhotoResourceUri": null,
              "interviewCount": 4,
              "isContagious": false,
              "isSuicidal": false,
              "isUnderMedication": false,
              "lastName": "Nordic",
              "location": "",
              "medicalCount": 5,
              "mustBeKeptAlone": false,
              "propertyBagNumber": null,
              "telephoneAcceptedCount": 6,
              "telephoneDeclinedCount": 7,
              "withCaution": false,
            }
          }
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        sm={12}
      >
        <Connect(WithStyles(DetaineeActionComponent))
          detainee={
            Object {
              "bailHearingCount": 0,
              "breathTestCount": 1,
              "cautionsArray": Array [
                "",
              ],
              "detentionUnitName": "Detainee Management Unit",
              "dob": "2001-01-01",
              "fingerPrintCount": 3,
              "firstName": "Steveo",
              "gender": "M",
              "hasWarning": false,
              "id": "123",
              "intakePhotoResourceUri": null,
              "interviewCount": 4,
              "isContagious": false,
              "isSuicidal": false,
              "isUnderMedication": false,
              "lastName": "Nordic",
              "location": "",
              "medicalCount": 5,
              "mustBeKeptAlone": false,
              "propertyBagNumber": null,
              "telephoneAcceptedCount": 6,
              "telephoneDeclinedCount": 7,
              "withCaution": false,
            }
          }
        />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(DetaineeGridComponent)>
</Fragment>
`;
